app/models/order.rb

```diff
@@ -0,0 +1,72 @@
+class Order < ApplicationRecord
+  belongs_to :user
+  has_many :order_books
+  has_many :books, through: :order_books
+
+  extend Enumerize
+  enumerize :status, in: Constants::ORDER_STATUS, predicates: true, scope: true
+
+  validates :shipping_address, presence: true
+  validates :total_price, presence: true
+
+  before_create :generate_id
+  after_update_commit :send_emails_for_confirmed_order, if: :fresh_confirmed?
+  after_update_commit :send_emails_for_cancelled_order, if: :fresh_cancelled?
```

Comment:
Generally, it's better to avoid callback like this. You can try creating a service object to handle order checkout, and put this email sending inside that service object.

Reply:
So, I should remove this callback, create a service object and call it in the controller. Is it right?

Reply:
Yes, it's something like that.

Reply:
Why do you think we need to avoid these callbacks? In case, we have a cms page, the admin can change the status of an order. we can use callbacks to trigger and send emails when the status is changed instead call the service in each function. I think it can help to not repeat the code.

Reply:
The reason is that we don't always want to trigger email when order is updated (like when admin changes status in CMS - sometimes we want to send email, sometimes we don't). If we want to DRY the code in those cases, we can still use the same service object.